src/jinja2/filters.py

```diff
@@ -410,7 +410,7 @@ def do_sort(
 
 
 @pass_environment
-def do_unique(
+def sync_do_unique(
```

Comment:
This is something I only learned today, so there may be downsides to it that I'm not yet aware of (other than making it more difficult to call the method from an interpreter):

Since the original `do_unique` method is _mostly_ async-compatible, it's possible to add the `async` keyword to it and change the `for item in value` loop to `async for item in value`.

That could avoid the addition of an `async_variant` decoration (and a small amount of method signature duplication).

Reply:
This is an API breakag for "sync" version of do_unique as caller will require to also use `async for x in list`.

Reply:
That made sense to me initially, but now I'm not so sure.  Is `do_unique` exposed as part of Jinja's API?

(templates containing `unique` as a filter called in a synchronous context continue to function when the `do_unique` method is declared `async`)

Reply:
The issue is not that you declare it `async`, but that you will return a `AsyncIterable` instead of a `Iterable`. All other filter expected as input an Iterable, not an AsyncIterable. This is where the API breakage is.

Reply:
Thank you; it took a while for me to return to this and to confirm the behaviour.  I frequently get confused about interface compatibility between async and non-async components.  Your approach in this pull request looks good to me.